lib/membrane/video_compositor/queue/offline/element.ex

```diff
@@ -115,92 +106,95 @@ defmodule Membrane.VideoCompositor.Queue.Offline.Element do
     {[], state}
   end
 
-  @spec next_interval_end(State.t()) :: Time.non_neg_t()
-  defp next_interval_end(%State{
-         previous_interval_end_pts: previous_interval_end_pts,
+  @spec calculate_next_buffer_pts(State.t()) :: Time.non_neg_t()
+  defp calculate_next_buffer_pts(%State{
+         next_buffer_pts: previous_buffer_pts,
          target_fps: {fps_num, fps_den}
        }) do
-    previous_interval_end_pts + Kernel.ceil(1_000_000_000 * fps_den / fps_num)
+    previous_buffer_pts + Kernel.ceil(Time.seconds(1) * fps_den / fps_num)
   end
 
-  @spec frame_or_eos?(list(PadState.pad_event())) ::
-          {queue_have_frame_or_eos? :: boolean(), event_type :: :frame | :end_of_stream}
+  @spec frame_or_eos?(list(PadState.pad_event())) :: :neither_frame_nor_eos | :frame | :eos
```

Comment:
I believe the convention says that the functions with names ending with `?` should return boolean 

```suggestion
  @spec frame_or_eos(list(PadState.pad_event())) :: :neither_frame_nor_eos | :frame | :eos
```